scripts/setup_ubuntu.sh

```diff
@@ -22,7 +22,6 @@ VERSION=$(lsb_release --release --short | cut -d . -f 1) # numerical major versi
 SETUP_TOOLCHAIN_VARIANTS=${SETUP_TOOLCHAIN_VARIANTS:-yes}
```

Comment:
I support these changes, but they are out of place in the PR. Up to you.

Reply:
You mean the addition of `libnuma-dev`? Let me put this as a separate commit. The reason it came up was that I was following our build instructions in `Python.md` and it mentioned `libnuma-dev` but I switched the instructions to reference `scripts/setup_ubuntu.sh` for simplicity, so it made sense to at least care `libnuma-dev` to `setup_ubuntu.sh`.

Reply:
Ah. I see. I actually cannot find a reference to this script. I do see `scripts/setup_dev_ubuntu.sh`, but I would rather not use that for conda builds. It installs a HUGE amount of extra stuff that is not needed for the conda build and does so at the system level. It also initializes conan which you should NEVER use with conda, so I would rather not encourage people to use it.

All that said, not having numa setup correctly doesn't prevent a correct build. It just prevents NUMA from working correctly at runtime.

Reply:
I suspect the Python build instructions could use some tweaking anyway. So I can do that in a separate PR. So continue as you see fit.

Reply:
`setup_ubuntu.sh` is referenced from `setup_dev_ubuntu.sh`.

```
katana/scripts/setup_dev_ubuntu.sh
28:sudo bash -x "${REPO_ROOT}/scripts/setup_ubuntu.sh" --no-setup-toolchain-variants
```

I'm going to leave libnuma in `setup_ubuntu.sh` because I deleted from Python.md. But yeah, feel free to streamline the `setup_dev_ubuntu.sh` for conda versus conan.